src/ccnl-fwd/src/ccnl-fwd.c

```diff
@@ -107,16 +111,23 @@ ccnl_fwd_handleContent(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
         return 0;
     }
 
-    if (!ccnl_content_serve_pending(relay, c)) { // unsolicited content
+    ccnl_prefix_to_str((*pkt)->pfx, s, CCNL_MAX_PREFIX_SIZE);
+    tclass = qos_traffic_class(s);
+    int pit_pending = ccnl_content_serve_pending(relay, c);
+    if (!pit_pending && !tclass->reliable) { // unsolicited content
```

Comment:
Previously, we would immediately quit out here if the content is unsolicited. Changed this so that unsolicited content can still be added to cache if it's marked as reliable. It will still not be forwarded (as this would happen in `ccnl_content_serve_pending()`, which already has logic preventing this).